packages/api/prisma/schema.prisma

```diff
@@ -57,8 +57,8 @@ enum Role {
 }
 
 model CourseProgress {
-  id                                String @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
-  courseUserId                      String @unique
-  learningMaterialsId               String @unique
-  learningMaterialsCompletedTasks   Int
+  id                              String  @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
+  courseUserId                    String?
```

Comment:
UNIQUE and NOT NULL for database are not exlusive. So you can have
`courseUserId String @unique`